ArduCopter/control_autonomous.cpp

```diff
@@ -257,14 +257,15 @@ bool Copter::autonomous_controller(float &target_climb_rate, float &target_roll,
 	target_pitch = 100.0f * g.pid_pitch.get_pid();
 
     // If we can move forward and we take the oppurtunity and try to hold horizontal pos
+    // CY: if the wall is 10 away, param1 is 50, the abs(diff) is still > 20 but drone would cont. to go forward, which is undesirable
     if(abs(g.e100_param1 - dist_forward) > 20 && state!=MOVING_FORWARD)
     {
         state = MOVING_FORWARD;
         moveForwardRightHold = dist_right;
         moveForwardLeftHold = dist_left;
     }
     // When we get out of moving forward, center again
-    else if(abs(g.e100_param1 - dist_forward) < 20 && state == MOVING_FORWARD)
+    else if(abs(g.e100_param1 - dist_forward) < 20 && state == MOVING_FORWARD)  //CY: maybe change this to an else
```

Comment:
If we make this else it would set state to HOLD_CENTER as long as we are close enough to a front wall. The purpose of this if is to check whether we were just moving forward and have reached the wall we were moving towards and then we want to readjust to center.